scompose/project/project.py

```diff
@@ -166,10 +166,30 @@ def parse(self):
                     working_dir=self.working_dir,
                 )
 
+            self.instances = self._sort_instances(self.instances)
+
             # Update volumes with volumes from
             for _, instance in self.instances.items():
                 instance.set_volumes_from(self.instances)
 
+    def _sort_instances(self, instances):
+        """eventually reorder instances based on depends_on constraints"""
+        sorted_instances = []
+        for instance in self.instances:
+            depends_on = self.instances[instance].params.get("depends_on", [])
+
+            try:
+                index = sorted_instances.index(instance)
+            except ValueError:
+                sorted_instances.append(instance)
+                index = sorted_instances.index(instance)
+
+            for dep in depends_on:
+                if not dep in sorted_instances:
+                    sorted_instances.insert(index, dep)
+
+        return {k: self.instances[k] for k in sorted_instances}
```

Comment:
Uhm, that way you are saying that whenever the function is called without names, we should provide the instances in reverse order. Isn't it a risky assumption? I would rather let instances always to be correctly ordered based on `depends_on` and just delegate the reverse to the `down` function.   

Regarding `OrderedDict`, I just replied in the next comment.

Reply:
I didn't say to add names.reverse() to the function to get the names, I suggested just using OrderedDict, so _after_ that is called, when we call reverse() it would be in reverse order. But you've clarified that for Python 3.7 and above, the dictionary is already honored to be ordered by insertion, so the reverse() would work fine here. Does this clear up your misread of my comment?

Reply:
> Does this clear up your misread of my comment?

Yeah, definitely! I had wrongly associated your comment to the quoted piece of code